hphp/hsl/src/str/introspect_l.php

```diff
@@ -0,0 +1,296 @@
+  Locale\Locale $locale,
+  string $string,
+  string $suffix,
+)[]: bool {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\ends_with_ci_l($string, $suffix, $locale);
+}
+
+/**
+ * Returns the length of the given string in characters.
+ *
+ * @see `Str\length()` (or pass `Locale\c()`) for the length in bytes.
+ */
+function length_l(
+  Locale\Locale $locale,
+  string $string,
+)[]: int {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\strlen_l($string, $locale);
+}
+
+/**
+ * Returns the first position of the "needle" string in the "haystack" string,
+ * or null if it isn't found.
+ *
+ * An optional offset determines where in the haystack the search begins. If the
+ * offset is negative, the search will begin that many characters from the end
+ * of the string. If the offset is out-of-bounds, a ViolationException will be
+ * thrown.
+ *
+ * - To simply check if the haystack contains the needle, see `Str\contains_l()`.
+ * - To get the case-insensitive position, see `Str\search_ci_l()`.
+ * - To get the last position of the needle, see `Str\search_last_l()`.
+ */
+function search_l(
+  Locale\Locale $locale,
+  string $haystack,
+  string $needle,
+  int $offset = 0,
+)[]: ?int {
+  /* HH_FIXME[4390] missing [] */
+  $position = _Str\strpos_l($haystack, $needle, $offset, $locale);
+  if ($position < 0) {
+    return null;
+  }
+  return $position;
+}
+
+/**
+ * Returns the first position of the "needle" string in the "haystack" string,
+ * or null if it isn't found (case-insensitive).
+ *
+ * Locale-specific rules for case-insensitive comparisons will be used.
+ *
+ * An optional offset determines where in the haystack the search begins. If the
+ * offset is negative, the search will begin that many characters from the end
+ * of the string. If the offset is out-of-bounds, a ViolationException will be
+ * thrown.
+ *
+ * - To simply check if the haystack contains the needle, see `Str\contains()`.
+ * - To get the case-sensitive position, see `Str\search()`.
+ * - To get the last position of the needle, see `Str\search_last()`.
+ */
+function search_ci_l(
+  Locale\Locale $locale,
+  string $haystack,
+  string $needle,
+  int $offset = 0,
+)[]: ?int {
+  /* HH_FIXME[4390] missing [] */
+  $position = _Str\stripos_l($haystack, $needle, $offset, $locale);
+  if ($position < 0) {
+    return null;
+  }
+  return $position;
+}
+
+/**
+ * Returns the last position of the "needle" string in the "haystack" string,
+ * or null if it isn't found.
+ *
+ * An optional offset determines where in the haystack (from the beginning) the
+ * search begins. If the offset is negative, the search will begin that many
+ * characters from the end of the string and go backwards. If the offset is
+ * out-of-bounds, a ViolationException will be thrown.
+ *
+ * - To simply check if the haystack contains the needle, see `Str\contains()`.
+ * - To get the first position of the needle, see `Str\search()`.
+ *
+ * Previously known in PHP as `strrpos`.
```

Comment:
Are these `Previously known in PHP as ...` comments accurate? I would have expected an `mb_` function here.